src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -0,0 +1,189 @@
+
+    private final UserRepository userRepository;
+
+    private final ConversationService conversationService;
+
+    public ChannelService(ConversationParticipantRepository conversationParticipantRepository, ChannelRepository channelRepository, UserRepository userRepository,
+            ConversationService conversationService) {
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.channelRepository = channelRepository;
+        this.userRepository = userRepository;
+        this.conversationService = conversationService;
+    }
+
+    /**
+     * Grans the channel moderator role to the given user for the given channel
+     *
+     * @param channel      the channel
+     * @param usersToGrant the users to grant the channel moderator role
+     */
+    public void grantChannelModeratorRole(Channel channel, Set<User> usersToGrant) {
+        var matchingParticipants = conversationParticipantRepository.findConversationParticipantsByConversationIdAndUserIds(channel.getId(),
+                usersToGrant.stream().map(User::getId).collect(Collectors.toSet()));
+        for (ConversationParticipant conversationParticipant : matchingParticipants) {
+            conversationParticipant.setIsModerator(true);
+        }
+        conversationParticipantRepository.saveAll(matchingParticipants);
+        conversationService.notifyAllConversationMembersAboutUpdate(channel);
+    }
+
+    /**
+     * Revokes the channel moderator role from a user for the given channel
+     *
+     * @param channel       the channel
+     * @param usersToRevoke the users to revoke channel moderator role from
+     */
+    public void revokeChannelModeratorRole(Channel channel, Set<User> usersToRevoke) {
+        var matchingParticipants = conversationParticipantRepository.findConversationParticipantsByConversationIdAndUserIds(channel.getId(),
+                usersToRevoke.stream().map(User::getId).collect(Collectors.toSet()));
+        for (ConversationParticipant conversationParticipant : matchingParticipants) {
+            conversationParticipant.setIsModerator(false);
+        }
+        conversationParticipantRepository.saveAll(matchingParticipants);
+        conversationService.notifyAllConversationMembersAboutUpdate(channel);
+    }
+
+    /**
+     * Updates the given channel
+     *
+     * @param channelId  the id of the channel to update
+     * @param courseId   the id of the course the channel belongs to
+     * @param channelDTO the dto containing the new channel data
+     * @return the updated channel
+     */
+    public Channel updateChannel(Long channelId, Long courseId, ChannelDTO channelDTO) {
+        var channel = channelRepository.findByIdElseThrow(channelId);
+        if (channelDTO.getName() != null && !channelDTO.getName().equals(channel.getName())) {
+            channel.setName(channelDTO.getName().trim().isBlank() ? "" : channelDTO.getName().trim());
```

Comment:
Hmm... why the distinction and not simply `channel.setName(channelDTO.getName().trim())`?

Reply:
True! Changed it in commit: b458ed39de275ed1eb7057c8a9e68610a94f6d31